### PR TITLE
Generate slugs from club titles in JSON file and append to each club entry

### PR DESCRIPTION
**Pull request by malcolmseyd**
This PR addresses issue #44.

A script was written to generate a slug for each club. This script was run on our JSON file.

Each slug contains only lowercase characters, numbers, and dashes. Some characters are replaced with their English equivalent ([the full list is here](https://github.com/simov/slugify/blob/1142e000f2b99552afb13d4118acbc25177df140/config/charmap.json)).

### CODE REVIEW

File: src/JSON/ClubData.json [hunk 1]
```diff
@@ -116,8 +123,9 @@
     "tags": [
       "Community Service & Volunteer"
     ],
-    "description": "Students Offering Support (SOS) is a nation-wide charity that supports university chapters across Canada. In addition to general fundraising, we host Exam-Aid preparation services to fund SOS Global Impact Projects in developing countries. SOS is a unique social enterprise that relies on passionate student leaders to create a positive impact, both at home and abroad, through the SOS model of \u201cRaising Marks, Raising Money, Raising Roofs.\u201d",
-    "createdDate": "1920-01-15 14:38:49"
+    "description": "Students Offering Support (SOS) is a nation-wide charity that supports university chapters across Canada. In addition to general fundraising, we host Exam-Aid preparation services to fund SOS Global Impact Projects in developing countries. SOS is a unique social enterprise that relies on passionate student leaders to create a positive impact, both at home and abroad, through the SOS model of “Raising Marks, Raising Money, Raising Roofs.”",
+    "createdDate": "1920-01-15 14:38:49",
+    "slug": "uvic-students-offering-support-(sos)"
```
Comment (aomi) [inline src/JSON/ClubData.json]: Should probably do something about brackets too. Could remove the brackets and contents altogether or maybe just remove brackets.

File: src/JSON/ClubData.json [hunk 2]
```diff
@@ -342,8 +364,9 @@
     "tags": [
       "Cultural"
     ],
-    "description": "UMANG: UVic Indian Students\u2019 Association was initially formed in 2005 by undergraduate and graduate students. We aim to promote Indian culture on campus and bridge cultural gaps in the society. Over the past few years, our team has grown and succeeded in hosting events and workshops. By doing so, we have promoted cultural diversity among students and community.We are one of the most active clubs on campus and organize events on a weekly basis.",
-    "createdDate": "1920-01-15 14:38:49"
+    "description": "UMANG: UVic Indian Students’ Association was initially formed in 2005 by undergraduate and graduate students. We aim to promote Indian culture on campus and bridge cultural gaps in the society. Over the past few years, our team has grown and succeeded in hosting events and workshops. By doing so, we have promoted cultural diversity among students and community.We are one of the most active clubs on campus and organize events on a weekly basis.",
+    "createdDate": "1920-01-15 14:38:49",
+    "slug": "umang-uvic-indian-students'-association"
```
Comment (aomi) [inline src/JSON/ClubData.json]: Quoting

File: src/JSON/ClubData.json [hunk 3]
```diff
@@ -358,7 +381,8 @@
       "Cultural"
     ],
     "description": "Society for Indian Classical Arts (SICA) is a not for profit student club at University of Victoria (UVic). SICA comprises of students, faculty and community members dedicated to promote and create awareness of South Asian arts and culture. Each year SICA brings numerous artists from across Canada who share their talent in various South Asian art forms. SICA has organized numerous events such as Indian Classical, Folk dances and music concerts.",
-    "createdDate": "1920-01-15 14:38:49"
+    "createdDate": "1920-01-15 14:38:49",
+    "slug": "society-for-indian-classical-arts-(sica)"
```
Comment (aomi) [inline src/JSON/ClubData.json]: Same deal with brackets here.

File: src/JSON/ClubData.json [hunk 4]
```diff
@@ -388,7 +413,8 @@
       "Cultural"
     ],
     "description": "We are Japanese students and those interested in Japanese culture. Our activities include language exchange for all levels of ability, social outings*, workshops, and anything you might like to see included. We welcome all students! Note: Due to Covid19 all social outings and in person events are post-poned",
-    "createdDate": "1920-01-15 14:38:49"
+    "createdDate": "1920-01-15 14:38:49",
+    "slug": "japanese-culture-club"
```
Comment (aomi) [inline src/JSON/ClubData.json:417]: Nice

File: src/JSON/ClubData.json [hunk 5]
```diff
@@ -403,7 +429,8 @@
       "Cultural"
     ],
     "description": "The UVicUSS is dedicated to promoting Ukrainian culture on campus through events, activism and community engagement. We have strong ties with the local Ukrainian community and are always excited to share our traditions with our peers. Whether they are Ukrainian or not, we welcome all students to come and enjoy what we have to offer!",
-    "createdDate": "1920-01-15 14:38:49"
+    "createdDate": "1920-01-15 14:38:49",
+    "slug": "uvic-ukrainian-students'-society"
```
Comment (aomi) [inline src/JSON/ClubData.json]: Quotation

File: src/JSON/ClubData.json [hunk 6]
```diff
@@ -748,7 +797,8 @@
       "Interest"
     ],
     "description": "We simulate the United Nations at various conferences around Canada and the US. We give students the chance to meet new people, learn new skills and to travel to new places!",
-    "createdDate": "1920-01-15 14:38:49"
+    "createdDate": "1920-01-15 14:38:49",
+    "slug": "model-united-nations-2"
```
Comment (aomi) [inline src/JSON/ClubData.json:801]: Is there legitimately two or do we have duplicate data?

File: src/JSON/ClubData.json [hunk 7]
```diff
@@ -778,7 +829,8 @@
       "Interest"
     ],
     "description": "The focus of this club is to help new brewers learn how to make not just beer, but other fermented beverages as well. We have started looking at techniques from around the world! From Korean rice wines, to Scandinavian farmhouse ales. Have you always wondered how these drinks were made? Join UVic Wizards of Beer and discover for yourself, and brew them too!",
-    "createdDate": "1920-01-15 14:38:49"
+    "createdDate": "1920-01-15 14:38:49",
+    "slug": "beer-club-(uvic-wizards-of-beer)"
```
Comment (aomi) [inline src/JSON/ClubData.json]: brackets

File: src/JSON/ClubData.json [hunk 8]
```diff
@@ -958,7 +1021,8 @@
       "Other"
     ],
     "description": "The UVFAC is a club dedicated to learning about investments for all skill levels. A place where students can ask questions and talk with others that are interested in a finance career or personal investing. All club funding is put directly towards financial learning tools. Please feel free to reach out at any time to our admins and come to our bi-weekly online meetings.",
-    "createdDate": "1920-01-15 14:38:49"
+    "createdDate": "1920-01-15 14:38:49",
+    "slug": "uvic-financial-analysis-club-(uvfac)"
```
Comment (aomi) [inline src/JSON/ClubData.json]: Brackets

File: src/JSON/ClubData.json [hunk 9]
```diff
@@ -1198,7 +1277,8 @@
       "Religious"
     ],
     "description": "We reach out to everyone who wants to know and discover the fullness of the Catholic faith, and we are dedicated to building up leaders who will change the world. We also provide opportunities for people to be engaged in a welcoming, loving community; with opportunities for spiritual, and personal growth. Join us for faith studies, events, Zoom hangouts, and retreats!",
-    "createdDate": "1920-01-15 14:38:49"
+    "createdDate": "1920-01-15 14:38:49",
+    "slug": "catholic-christian-outreach-(cco)"
```
Comment (aomi) [inline src/JSON/ClubData.json]: Brackets

File: src/JSON/ClubData.json [hunk 10]
```diff
@@ -1273,8 +1357,9 @@
     "tags": [
       "Religious"
     ],
-    "description": "The ABS club (Association for Baha\u2019i studies) is a club that allows Baha\u2019i students and faculty (as well as any one else interested) to aid one another in meaningful service to our campus community as well as reflect on how we can coherently integrate our spiritual beliefs and academic study. This club hopes to hold online social spaces to increase a sense of community and engage in conversation around spiritual topics.",
-    "createdDate": "1920-01-15 14:38:49"
+    "description": "The ABS club (Association for Baha’i studies) is a club that allows Baha’i students and faculty (as well as any one else interested) to aid one another in meaningful service to our campus community as well as reflect on how we can coherently integrate our spiritual beliefs and academic study. This club hopes to hold online social spaces to increase a sense of community and engage in conversation around spiritual topics.",
+    "createdDate": "1920-01-15 14:38:49",
+    "slug": "abs-(association-for-baha'i-studies)"
```
Comment (aomi) [inline src/JSON/ClubData.json]: Brackets and quotation

File: src/JSON/ClubData.json [hunk 11]
```diff
@@ -1288,8 +1373,9 @@
     "tags": [
       "Religious"
     ],
-    "description": "TharAllah: Shia Muslim Society of Victoria\u201d is a religious club intended to commemorate important dates in the Shia-Islamic calendar by holding events regularly as well as on special occasions for our growing community here on campus. TharAllah provides an opportunity for students from Muslim and other backgrounds to get together over religious and cultural programmes, socialize, and take part in developing our community.",
-    "createdDate": "1920-01-15 14:38:49"
+    "description": "TharAllah: Shia Muslim Society of Victoria” is a religious club intended to commemorate important dates in the Shia-Islamic calendar by holding events regularly as well as on special occasions for our growing community here on campus. TharAllah provides an opportunity for students from Muslim and other backgrounds to get together over religious and cultural programmes, socialize, and take part in developing our community.",
+    "createdDate": "1920-01-15 14:38:49",
+    "slug": "tharallah-(shia-muslim-society-of-victoria)"
```
Comment (aomi) [inline src/JSON/ClubData.json]: Brackets

File: src/JSON/ClubData.json [hunk 12]
```diff
@@ -1304,7 +1390,8 @@
       "Religious"
     ],
     "description": "As a religious group, our goal is to gather new and old UVic ismaili students for prayers. We provide a room where they can come together to study, pray and just hangout for short events. Besides officiating religious ceremonies on a daily basis, we also intend to have a mentorship programs, career panels, fund raisings and other events throughout the term. This year all the activities are virtual by getting together with SFU and UBC ISAs.",
-    "createdDate": "1920-01-15 14:38:49"
+    "createdDate": "1920-01-15 14:38:49",
+    "slug": "uvic-ismaili-students-association-2"
```
Comment (aomi) [inline src/JSON/ClubData.json]: Possible duplicate data?

File: src/JSON/ClubData.json [hunk 13]
```diff
@@ -1515,7 +1615,8 @@
       "STEM"
     ],
     "description": "BMED is a group of undergraduate engineering students who design, research, and test medical devices, biomedical technologies, and assistive solutions.",
-    "createdDate": "1920-01-15 14:38:49"
+    "createdDate": "1920-01-15 14:38:49",
+    "slug": "biomedical-engineering-design-team-(bmed)"
```
Comment (aomi) [inline src/JSON/ClubData.json]: Brackets

File: src/JSON/ClubData.json [hunk 14]
```diff
@@ -1740,7 +1855,8 @@
       "STEM"
     ],
     "description": "SUCO provides free consulting to STEM based organizations/companies and hosts free seminars presented by professionals in the STEM field on a vast array of topics.",
-    "createdDate": "1920-01-15 14:38:49"
+    "createdDate": "1920-01-15 14:38:49",
+    "slug": "stem-undergraduate-consulting-organization-(suco)"
```
Comment (aomi) [inline src/JSON/ClubData.json]: Brackets

File: scripts/slugify.js [hunk 15]
```diff
@@ -0,0 +1,34 @@
...
+
+async function main() {
+  const originalBytes = (await fs.readFile(inputFilePath)).buffer;
+  const originalString = new Buffer.from(originalBytes).toString();
+  const originalJSON = JSON.parse(originalString);
+
+  const sluggedJSON = originalJSON.map(slugMap);
+
+  console.log(JSON.stringify(sluggedJSON, null, 2));
+}
```
Comment (aomi) [inline scripts/slugify.js]: The `.nvmrc` is v14 so top-level await is a thing

File: src/JSON/ClubData.json (hunk 1 above)
Comment (malcolmseyd) [reply src/JSON/ClubData.json]: addressed in 9983d3e

File: src/JSON/ClubData.json (hunk 2 above)
Comment (malcolmseyd) [reply src/JSON/ClubData.json]: addressed in 9983d3e

File: src/JSON/ClubData.json (hunk 3 above)
Comment (malcolmseyd) [reply src/JSON/ClubData.json]: addressed in 9983d3e

File: src/JSON/ClubData.json (hunk 4 above)
Comment (malcolmseyd) [reply src/JSON/ClubData.json:417]: nice

File: src/JSON/ClubData.json (hunk 5 above)
Comment (malcolmseyd) [reply src/JSON/ClubData.json]: addressed in 9983d3e

File: src/JSON/ClubData.json (hunk 11 above)
Comment (malcolmseyd) [reply src/JSON/ClubData.json]: addressed in 9983d3e

File: src/JSON/ClubData.json (hunk 13 above)
Comment (malcolmseyd) [reply src/JSON/ClubData.json]: addressed in 9983d3e

File: src/JSON/ClubData.json (hunk 14 above)
Comment (malcolmseyd) [reply src/JSON/ClubData.json]: addressed in 9983d3e

File: src/JSON/ClubData.json (hunk 12 above)
Comment (malcolmseyd) [reply src/JSON/ClubData.json]: addressed in e1458ef

File: scripts/slugify.js (hunk 15 above)
Comment (malcolmseyd) [reply scripts/slugify.js]: addressed in f4eca31

**Comment by malcolmseyd** [comment]
I addressed all review comments, please re-review.

File: src/JSON/ClubData.json (hunk 6 above)
Comment (aomi) [reply src/JSON/ClubData.json:801]: @malcolmseyd I can't imagine we have two model UN clubs at UVic?

**Review by aomi — APPROVED** [review]
LGTM subject to pending duplication entry review

**Pull request merged**